File: src/main/kotlin/net/ponec/demo/exposed/controller/DemoRest.java
package net.ponec.demo.exposed.controller;

import net.ponec.demo.exposed.entity.Department;
import net.ponec.demo.exposed.entity.Employee;
import net.ponec.demo.exposed.entity.EmployeeDto;
import net.ponec.demo.exposed.service.DataService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.Collections;
import java.util.List;

@RestController("/")
public class DemoRest {

    private final DataService dataService;

    public DemoRest(DataService dataService) {
        this.dataService = dataService;
    }

    @GetMapping("hello")
    public String hello(@RequestParam(required = false) String name) {
        return "Hallo " + name;
    }

    /** URL: http://localhost:8080/employees */
    @GetMapping(value = "employees")
    public List<EmployeeDto> findAllEmployees() {
        return dataService.findAllEmployees();
    }


}

File: src/main/kotlin/net/ponec/demo/controller/DemoRest.kt
package net.ponec.demo.controller

import net.ponec.demo.exposed.service.ExposedService
import net.ponec.demo.ktorm.service.KtormService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController("/")
class DemoRest(
    private val exposedService: ExposedService,
    private val ktormService: KtormService

) {
    @GetMapping("hello")
    fun hello(@RequestParam(required = false) name: String): String {
        return "Hallo $name"
    }

    /** Call an Exposed service.
     * URL: http://localhost:8080/employeesExpByEntity */
    @GetMapping(value = ["employeesExpByEntity"])
    fun findAllEmployeesByEntity(): List<EmployeeDto> {
        return exposedService.findAllEmployeesByEntity()
    }

    /** Call an Exposed service.
     * URL: http://localhost:8080/employeesExpByTable */
    @GetMapping(value = ["employeesExpByTable"])
    fun findAllEmployeesByTable(): List<EmployeeDto> {
        return exposedService.findAllEmployeesByTable()
    }

    /** Call a Ktorm service.
     * URL: http://localhost:8080/employeesKtormByEntity  */
    @GetMapping(value = ["employeesKtormByEntity"])
    fun findAllEmployees2(): List<EmployeeDto> {
        return ktormService.findAllEmployeesByEntity()
    }

    /** Call a Ktorm service.
     * URL: http://localhost:8080/employeesKtormByTable  */
    @GetMapping(value = ["employeesKtormByTable"])
    fun findAllEmployees3(): List<EmployeeDto> {
        return ktormService.findAllEmployeesByTable()
    }
}